server/schedule/plan/status.go

```diff
@@ -0,0 +1,120 @@
+	StatusOK StatusCode = iota
+	// StatusNoNeed represents the plan is no need to be scheduled.
+	StatusNoNeed
+	// StatusPaused represents the scheduler or checker is paused.
+	StatusPaused
+
+	// StatusStoreThrottled represents the store cannot be selected due to the limitation.
+	StatusStoreThrottled
+	// StatusStoreUnavailable represents the store cannot be selected due to it's state.
+	StatusStoreUnavailable
+	// StatusStoreLowSpace represents the store cannot be selected because it runs out of space.
+	StatusStoreLowSpace
+	// StatusStoreDraining represents the data on the store is moving out.
+	StatusStoreDraining
+	// StatusStoreBlocked represents the store is restricted by the special configuration.
+	StatusStoreBlocked
+	// StatusStoreExcluded represents the store is excluded due to the existed or unhealthy region peer.
+	StatusStoreExcluded
+
+	// StatusRegionHot represents the region cannot be selected due to the heavy load.
+	StatusRegionHot
+	// StatusRegionUnhealthy represents the region cannot be selected due to the region health.
+	StatusRegionUnhealthy
+	// StatusRegionEmpty represents the region cannot be selected due to the region is empty.
+	StatusRegionEmpty
+
+	// StatusLabelNotMatch represents the location label of placement rule is not match the store's label.
+	StatusLabelNotMatch
+	// StatusRuleNotMatch represents the placement rule cannot satisfy the requirement.
+	StatusRuleNotMatch
+	// StatusIsolationNotMatch represents the isolation cannot satisfy the requirement.
+	StatusIsolationNotMatch
+
+	// TODO: The below status is not used for now. Once it is used, Please remove this comment.
+
+	// StatusStoreNotExisted represents the store cannot be found in PD.
+	StatusStoreNotExisted
+	// StatusNoTargetRegion represents the target region of merge operation cannot be found.
+	StatusNoTargetRegion
+	// StatusRegionLabelReject represents the plan conflicts with region label.
+	StatusRegionLabelReject
+)
+
+var statusText = map[StatusCode]string{
+	StatusOK:     "OK",
+	StatusNoNeed: "No Need",
+	StatusPaused: "Paused",
+
+	// store
+	StatusStoreThrottled:   "Store Throttled",
+	StatusStoreUnavailable: "Store Unavailable",
+	StatusStoreLowSpace:    "Store Low Space",
+	StatusStoreDraining:    "Store Draining",
+	StatusStoreBlocked:     "Store Blocked",
+	StatusStoreExcluded:    "Region Excluded",
+
+	// region
+	StatusRegionHot:       "Region Hot",
+	StatusRegionUnhealthy: "Region Unhealthy",
+	StatusRegionEmpty:     "Region Empty",
+
+	StatusLabelNotMatch:     "Label Not Match",
+	StatusRuleNotMatch:      "Rule Not Match",
+	StatusIsolationNotMatch: "Isolation Not Match",
+
+	// non-filter
+	StatusStoreNotExisted:   "Store Not Existed",
+	StatusNoTargetRegion:    "No Target Region",
+	StatusRegionLabelReject: "Region Label Reject",
+}
+
+// StatusText turns the status code into string.
+func StatusText(code StatusCode) string {
+	return statusText[code]
+}
+
+// Status describes a plan's result.
+type Status struct {
```

Comment:
How about define a function( for example)`IsOK` for `Status`?  since there are so much 
```if filter.Target(c.cluster.GetOpts(), s).StatusCode != plan.StatusOK```
and with function `IsOK` are more convenient for the developer
```if filter.Target(c.cluster.GetOpts(), s).IsOK()```